File: CS/WindowsApplication1/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;

namespace Compass {
    public partial class Form1 : Form {
        public Form1() {
            InitializeComponent();
        }
        void OnCustomTickmarkText(DevExpress.XtraGauges.Core.Model.CustomTickmarkTextEventArgs ea) {
            string text = ea.Text;
            switch(ea.Text) {
                case "360": text = "-N-"; break;
                case "270": text = "-W-"; break;
                case "180": text = "-S-"; break;
                case "90": text = "-E-"; break;
            }
            ea.Result = text;
        }

		float ToAngle(int angle) {
            if(angle > 0) angle = angle % 360;
            else angle = 360 - Math.Abs(angle) % 360;
            return (float)angle;
        }

		private void trackBarControl1_EditValueChanged(object sender, EventArgs e) {
			this.arcScaleComponent1.Value = ToAngle(trackBarControl1.Value);

		}
    }
}

File: VB/WindowsApplication1/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms

Namespace Compass
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub OnCustomTickmarkText(ByVal ea As DevExpress.XtraGauges.Core.Model.CustomTickmarkTextEventArgs) Handles arcScaleComponent1.CustomTickmarkText
			Dim text As String = ea.Text
			Select Case ea.Text
				Case "360"
					text = "-N-"
				Case "270"
					text = "-W-"
				Case "180"
					text = "-S-"
				Case "90"
					text = "-E-"
			End Select
			ea.Result = text
		End Sub

		Private Function ToAngle(ByVal angle As Integer) As Single
			If angle > 0 Then
				angle = angle Mod 360
			Else
				angle = 360 - Math.Abs(angle) Mod 360
			End If
			Return CSng(angle)
		End Function

		Private Sub trackBarControl1_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles trackBarControl1.EditValueChanged
			Me.arcScaleComponent1.Value = ToAngle(trackBarControl1.Value)

		End Sub
	End Class
End Namespace